src/lib/content/coordinators/GetContentItemV2Impl.ts

```diff
@@ -0,0 +1,67 @@
+import { GetContentItemById } from './GetContentItemById';
+import { GetContentItemByKey } from './GetContentItemByKey';
+import { ContentBody } from '../model/ContentBody';
+import { ContentItem } from '../model/ContentItem';
+import { createContentClient } from '../../client/createContentClient';
+import { ContentMapper } from '../mapper/ContentMapper';
+import { AxiosInstance, AxiosResponse } from 'axios';
+import { encodeQueryString } from '../../utils/Url';
+import { ContentClientConfigV2 } from '../../config/ContentClientConfigV2';
+
+/**
+ * @hidden
+ */
+export class GetContentItemV2Impl
+  implements GetContentItemById, GetContentItemByKey {
+  private readonly contentClient: AxiosInstance;
+
+  constructor(
+    private readonly config: ContentClientConfigV2,
+    private readonly mapper: ContentMapper
+  ) {
+    this.contentClient = createContentClient(
+      this.config,
+      `https://${config.hubName}.cdn.content.amplience.net`
+    );
+  }
+
+  getContentItemByKey<T extends ContentBody>(
+    key: string
+  ): Promise<ContentItem<T>> {
+    return this.getContentItem('key', key);
+  }
+  async getContentItemById<T extends ContentBody>(
+    id: string
+  ): Promise<ContentItem<T>> {
+    return this.getContentItem('id', id);
+  }
+
+  async getContentItem<T extends ContentBody>(
+    requestType: 'id' | 'key',
+    value: string
+  ): Promise<ContentItem<T>> {
+    const args = [
+      ['depth', 'all'],
+      ['format', 'inlined'],
+    ];
+    if (this.config.locale) {
+      args.push(['locale', this.config.locale]);
+    }
+    const url = `content/${requestType}/${value}?${encodeQueryString(args)}`;
+
+    let response: AxiosResponse;
+    try {
+      response = await this.contentClient.get(url);
+    } catch (err) {
+      if (err.response && err.response.status === 404) {
```

Comment:
Not sure whether this adds anything useful and may just make error handling more work

Reply:
I guess my thinking was that a 404 error is an accepted use case in our SDK, so we can throw our own exception for this instead of the underlying exception. I suppose the axios error would contain more information, the full request, the full response, stack trace etc. However clients might start to rely on it it's not _our_ exception....

My preference would be replace axios with `fetch()`... I might start a R&D branch into that.